### ISSUE TITLE
CombiTables: Empty table fill(0.0, 0, 2) raises an assertion

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1137]
**Reported by beutlich on 28 May 2013 09:54 UTC**
A customer of SimulationX wondered why the default _table_=fill(0.0, 0, 2) of _CombiTimeTable_ and _CombiTablenD_ is an empty table that always raises the assertion
```
Assert failed: tableOnFile = false and parameter table is an empty matrix
```
if simulated without component modification. In comparison _TimeTable.table_ is set to [0, 0; 1, 1; 2, 4] by default such that simulation succeeds.

I suggest to also use a proper table matrix for the CombiTables.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1137

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 28 May 2013 10:04 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 May 2013 10:22 UTC**
Replying to [ticket:1137 beutlich]:
> A customer of SimulationX wondered why the default _table_=fill(0.0, 0, 2) of _CombiTimeTable_ and _CombiTablenD_ is an empty table that always raises the assertion
> ```
> Assert failed: tableOnFile = false and parameter table is an empty matrix
> ```
> if simulated without component modification. In comparison _TimeTable.table_ is set to [0, 0; 1, 1; 2, 4] by default such that simulation succeeds.
> 
> I suggest to also use a proper table matrix for the CombiTables.

We had a long discussion on default values in MSL and the preferred style is documented here: Modelica.UsersGuide.ParameterDefaults

To summarize, for parameters where a meaningful default cannot be given, the default should be given as start-value (there is the convention that a tool prints a warning when simulation starts, if start values are not modified).

The same should be done with the tables:

CombiTimeTable should be changed so that the default value is defined as start value and for the other tables start values should be provided. Can you do this?

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 May 2013 11:37 UTC**
Replying to [comment:2 otter]:
> CombiTimeTable should be changed so that the default value is defined as start value and for the other tables start values should be provided.

I guess you mean TimeTable here.
> Can you do this?

Yes, I will do it. Changing the table (start) value should not break the backward compatibility (for the very reason that current default asserts).

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 May 2013 12:38 UTC**
Resolved by 414466def2e5ead4621e14276d5247658fe67340.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 May 2013 13:41 UTC**
Dymola 2014 now warns on Modelica.Blocks.Examples.Interaction1.
```
Check of Modelica.Blocks.Examples.Interaction1:
Incompatible sizes for variable and its start value;
timeTable.table = [0, 0; 1, 1; 2, 4]
They have for dimension 1 the sizes 7 and 3.
Errors detected in variable declarations.
Check of class aborted.
WARNING: 1 warning was issued
```

SimulationX even fails
```
Error In timeTable.table.start
timeTable.table.start:=
                       noEvent([0,0;1,1;2,4]);
In an assignment, the variable ([7,2]-matrix) must have the same dimension as the expression ([3,2]-matrix).
```

Should the start values be changed (back) to parameter declarations?

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 May 2013 19:48 UTC**
Replying to [comment:5 beutlich]:


> Dymola 2014 now warns on Modelica.Blocks.Examples.Interaction1.
> ```
> Check of Modelica.Blocks.Examples.Interaction1:
> Incompatible sizes for variable and its start value;
> timeTable.table = [0, 0; 1, 1; 2, 4]
> They have for dimension 1 the sizes 7 and 3.
> Errors detected in variable declarations.
> Check of class aborted.
> WARNING: 1 warning was issued
> ```
> 
> Should the start values be changed (back) to parameter declarations?

Sorry you are right. This is not correct Modelica.

If a default table is given, then there is always the danger that a user "forgets" to set this parameter, and the tool cannot warn.

Therefore, it seems best to revert this change to the previous version (and in order to be backwards compatible keep the default table for TimeTable).

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 May 2013 20:10 UTC**
Fixed in c0f62028af0b89d3760cd0af6b9dece7c6acedeb:
Re-did the change and added zero-dimension tables as defaults for CombiTimeTables and removed start value (and introduced no default value) in TimeTable (as it was the case in MSL 3.2).

Set the flag to "Hide ticket" in order that it will not appear in the release notes.,

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 20 Dec 2016 12:19 UTC**
